### PR TITLE
Docker distribution

### PR DESCRIPTION
**Pull request by augi**
Please note that you must create a new repository in Docker Hub and setup TravisCI to pass arguments used to execute `docker login`.

### CODE REVIEW

**Comment by alenkacz** [comment]
@augi I would prefer to use docker hub builds instead of pushing image from travis. What do you think? I am already using it for rabbitmq scaler so I can set it up quite easily...

**Comment by augi** [comment]
I've never done it - so feel free to change this branch to use Docker Hub Builds.

File: README.md [hunk 1]
```diff
@@ -23,32 +23,46 @@ Usage
...
 Properties
-====================
-- *url* - url to your marathon application
-- *dockerImageName* (OPTIONAL) - full image name that will be provided to marathon, use this only if you want to override the image name provided in your json (e.g. you want to deploy a testing version as a part of your CI/CD pipeline)
-- *pathToJsonFile* (OPTIONAL) - project relative path to your json file, default is *deploy/marathon.json*
-- *verificationTimeout* (OPTIONAL) - timeout when querying Marathon to verify that there are no pending deployments left, default is 90 seconds
-- *deploymentRequestTimeout* (OPTIONAL) - timeout for the initial deployment request, default is 5 seconds
-- *jvmOverhead* (OPTIONAL) - number of MBs that the container needs on top of JVM app memory (default is 200)
+==========
+| Gradle property | Docker environment variable | Default | Description |
```
Comment (alenkacz) [inline README.md]: when previewing the table it does not look that good on my computer. The last column with the comments is too small :(

File: docker-distribution/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM gradle:3.5-jre8-alpine
+MAINTAINER varkockova.a@gmail.com
+
+USER root
+COPY build.gradle /
+WORKDIR /
+ARG PLUGIN_VERSION
+# ARGs are not persisted so we must store value of ARG to ENV
+ENV PLUGIN_VERSION=$PLUGIN_VERSION
```
Comment (alenkacz) [inline docker-distribution/Dockerfile:9]: I would prefer if this was a ARG and new tagged version of docker image will be released for every new release of the plugin. What do you think?

Comment (augi) [reply docker-distribution/Dockerfile:9]: This is how it works :) The problem is that ARG value is not persisted in the image and so we must store it to ENV.

Comment (alenkacz) [reply docker-distribution/Dockerfile:9]: not sure I get it. ENV is something that I provide when I run the image, right? I do not want user of this image to provide version of plugin in docker run...

**Pull request merged**